script.js

```diff
@@ -0,0 +1,149 @@
+    'Button 1 for Popup 1',
+    'Button 2 for Popup 1',
+  );
+});
+
+const popupButton2 = document.getElementById('popupButton2');
+popupButton2.addEventListener('click', () => {
+  openPopup(
+    'Multi-Post Stories',
+    ' ',
+    './images/Snapshoot 2.png',
+    './images/SnapshootPortfolio.svg',
+    'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essent',
+    'Button 1 for Popup 2',
+    'Button 2 for Popup 2',
+  );
+});
+
+const popupButton3 = document.getElementById('popupButton3');
+popupButton3.addEventListener('click', () => {
+  openPopup(
+    'TONIC',
+    ' ',
+    './images/Snapshoot 3.png',
+    './images/SnapshootPortfolio.svg',
+    'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essent',
+    'Button 1 for Popup 3',
+    'Button 2 for Popup 3',
+  );
+});
+
+const popupButton4 = document.getElementById('popupButton4');
+popupButton4.addEventListener('click', () => {
+  openPopup(
+    'Multi-Post Stories',
+    ' ',
+    './images/Snapshoot Portfolio 4.png',
+    './images/SnapshootPortfolio.svg',
+    'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essent',
+    'Button 1 for Popup 4',
+    'Button 2 for Popup 4',
+  );
+});
+
+closeButton.addEventListener('click', () => {
+  popupContainer.style.display = 'none';
+});
+
+const email = document.querySelector('#email');
+
+email.addEventListener('input', (event) => {
+  const correctInput = event.target.value.toLowerCase();
+  if (event.target.value !== correctInput) {
+    email.setCustomValidity('Please enter your email address using lowercase letters.');
+  } else {
+    email.setCustomValidity('');
+  }
+});
+
+// Check if there is any data in local storage
+if (localStorage.getItem('formData')) {
+  const savedData = JSON.parse(localStorage.getItem('formData'));
+  document.getElementById('name').value = savedData.name || '';
+  document.getElementById('email').value = savedData.email || '';
+  document.getElementById('mensage').value = savedData.mensage || '';
+}
```

Comment:
- [ ] As mentioned previously the value from the message input is still not being preserved upon page reloads. I've debugged the problem and have found that on line `139` you're selecting all `input` elements but the message field is a `textarea` element instead of an input so that means it's not being properly fetched from the `DOM`. You can try other alternatives like class selector by giving all the fields the same class name. 😉
- Here's how it behaves:  
  ![firefox_68SJtlerIp](https://github.com/jovasal/jovasal.github.io/assets/104968236/b89fec84-8868-42a5-9173-688f90cd7c92)

Reply:
Hi @AbelG101 .

Thank you so much for your recommendation.

I add:
const message = document.querySelector('#message');

and I made some changes. so now it is working fine!!!

I forgot to put a texarea. Thank you so much!!

Now is working fine!!!